File: app/app/src/main/java/tv/twitch/android/core/buildconfig/BuildConfigUtil.java
package tv.twitch.android.core.buildconfig;

import tv.orange.core.Hook;

public class BuildConfigUtil {
    /* ... */

    public final boolean isDebugConfigEnabled() { // TODO: __REPLACE_METHOD
        return Hook.inDevMode();
    }

    public final boolean isAlpha() { // TODO: __REPLACE_METHOD
        return Hook.inDevMode();
    }

    public final boolean isBeta() { // TODO: __REPLACE_METHOD
        return Hook.inDevMode();
    }

    public final boolean shouldShowDebugOptions(boolean z) { // TODO: __REPLACE_METHOD
        return Hook.inDevMode();
    }

    public final boolean bugReportingEnabled() { // TODO: __REPLACE_METHOD
        return Hook.inDevMode();
    }

    /* ... */
}

File: app/core/src/main/java/tv/orange/core/BuildConfigUtil.kt
package tv.orange.core

import org.json.JSONObject
import tv.orange.models.abc.OrangeBuildConfig
import tv.twitch.android.app.core.ApplicationContext
import java.util.*

object BuildConfigUtil {
    private const val BUILD_JSON_FILENAME = "build.json"
    private const val USER_AGENT_TEMPLATE = "PurpleTV/%s Build/%d"

    val userAgent by lazy {
        String.format(
            Locale.ENGLISH,
            USER_AGENT_TEMPLATE,
            buildConfig.getVersion(),
            buildConfig.number
        )
    }

    val buildConfig: OrangeBuildConfig by lazy {
        return@lazy try {
            with(JSONObject(getRawBuildString())) {
                OrangeBuildConfig(
                    number = getInt("number"),
                    version = getInt("version"),
                    timestamp = getInt("timestamp"),
                    sentryDNS = getString("sentryDNS"),
                    codename = getString("codename")
                )
            }
        } catch (e: Throwable) {
            OrangeBuildConfig()
        }
    }

    private fun getRawBuildString(): String {
        val context = ApplicationContext.getInstance().context

        return context.assets.open(BUILD_JSON_FILENAME).bufferedReader().use {
            it.readText()
        }
    }
}